modules/win-autoscale/watchmaker-win-autoscale.template.cfn.yaml

```diff
@@ -548,6 +560,36 @@ Resources:
                       "log_stream_name": "default_logs_{instance_id}"
                     }
                   }
+            'c:\cfn\scripts\AmazonCloudWatchAgent\cloudwatch-applog-config.ps1':
+              content: !Sub 
+                - |-
+                  $cloudwatch_baseline = 'c:/cfn/scripts/AmazonCloudWatchAgent/aws-cloudwatch-agent-config.json'
+                  $log_group_name = "/aws/ec2/win/${AWS::StackName}"
+                  $log_paths_input = "${local_addCWLtarget}".split(",")
+
+                  function get_stream_name($log_path){
+                    return (Split-Path -Leaf $log_path).split(".")[0] + "_logs_{instance_id}"
+                  }
+
+                  if($log_paths_input.count -gt 0){
+                    Write-Host "Additional cwl paths were defined"
+                    Write-Host "Loading existing cwl baseline"
+                    $json_obj = ConvertFrom-Json -InputObject (Get-Content $cloudwatch_baseline -RAW -Encoding ASCII)
+                    foreach ($log_path in $log_paths_input){
+                      $temp = [ordered]@{
+                      file_path=$log_path;
+                      log_group_name=$log_group_name;
+                      log_stream_name=get_stream_name($log_path);
+                      timestamp_format="%H:%M:%S %y %b %-d";
+                    }
+                    $json_obj.logs.logs_collected.files.collect_list += $temp
+                    }
+                  Write-Host "Writing updated cwl baseline."
+                  $json_obj | ConvertTo-Json -Depth 100 | Out-File -FilePath $cloudwatch_baseline -Encoding ASCII
+                  Write-Host "CWL baseline modification complete"
+                  }
+                - 
+                  local_addCWLtarget: !Join ["," ,!Ref CloudWatchAppLogs]
```

Comment:
Did I miss a note as to why we are doing powershell here instead of python? Thinking this _should_ be basically exactly the same between windows and linux in python...